### PR TITLE
Addon note without decorator

### PR DESCRIPTION
**Pull request by alexandrebodin**
## What I did

I changed the notes addon to be usable as a simple function. 

This brings some cool benefits:

- The user can call the function anytime during the story life cycle (e.g update notes on an event in the story)
- No need to wrap the story (cleaner for the user)

@shilman @ndelangen 

If you like it we can I'll merge into vue to bring it for the Vue release

Somewhat related, I'd like your thoughts on this [idea (gist)](https://gist.github.com/alexandrebodin/f470bb293e728be359bf739c9750ca15)
=> Make addon API available in props 

Cheers!

### CODE REVIEW

File: examples/vue/src/stories/index.js [hunk 1]
```diff
@@ -76,23 +76,53 @@ storiesOf('Other', module)
...
+    return {
+      template: '<div>A looooooooonnnnnnnggggggggggggg text</div>'
+    };
+  })
+  .add('with a button and clickable update', () => {
+    withNotes('My notes on a button and clickable update')
+    return {
+      methods: {
+        onClick() {
+          withNotes('My new notes');
```
Comment (shilman) [inline examples/vue/src/stories/index.js]: Interesting 😍

File: addons/notes/README.md [hunk 2]
```diff
@@ -37,5 +37,8 @@ import { withNotes } from '@storybook/addon-notes';
 import Component from './Component';
 
 storiesOf('Component', module)
-  .add('with some emoji', withNotes({ notes: 'A very simple component'})(() => <Component></Component>));
+  .add('with some emoji', () => {
+    withNotes('some notes')
```
Comment (shilman) [inline addons/notes/README.md]: @alexandrebodin minor nitpick, but this should have semicolon ... I think our linting is setup to fix this automatically and should be running before you can commit. I don't care too much about semiicolons, but am more concerned about bigger problems getting through

**Comment by shilman** [comment]
Hi @alexandrebodin , thanks for this.

### PROS:
- It's very clear/clean/simple
- It should work with any any framework we choose to support

### CONS:
- It doesn't address cases where the panel data is dependent (kind/story/props)
- It mixes concerns of "story" and "story decorators" which seems like a bad idea

@tmeasday has written up a proposal for an HOC-based approach that preserves the separation of story and decorator. Curious what you think about this:

https://gist.github.com/tmeasday/775c5572271121bb2a71153ac5f2cd7a

**Comment by tmeasday** [comment]
This is interesting!

I'd say that it's pretty unusual for an "inspector" addon to not actually need to see the rendered story, so I'm not sure in general we can learn too much from this example. 

I think there's an even simpler way to go in this case, though. If we support options in the `.add()` function, we can just do:

```js
.add(name, () => <story/>, { notes: "Here's my note" });
```

We don't get this ability to change notes as you interact with the story, but that feels a bit YAGNI, to me..

**Comment by ndelangen** [comment]
What's the progress on this PR? Is this something we will be merging at some point or mostly a discussion?

**Comment by alexandrebodin** [comment]
Only a discussion :)

**Comment by ndelangen** [comment]
All right, I hope you don't me closing it then, because we have quite a few PRs open.

**Pull request closed without merging**